webui/react/src/e2e/tests/experimentList.spec.ts

```diff
@@ -439,4 +440,126 @@ test.describe('Experiment List', () => {
+          const expectedAscendingValues = [...valuesToCompare].sort();
+          expect(expectedAscendingValues).toStrictEqual(valuesToCompare);
+        } else {
+          const expectedAscendingValues = [...valuesToCompare].sort((a, b) => {
+            if (a < b) return 1;
+            if (a > b) return -1;
+
+            return 0;
+          });
+          expect(expectedAscendingValues).toStrictEqual(valuesToCompare);
+        }
+      };
+      const checkTableOrder = async (firstKey: string, secondKey: string, descending = false) => {
+        const rowIndexes = (await projectDetailsPage.f_experimentList.dataGrid.allRows()).map(
+          (_, index) => index,
+        );
+        await validateByColumn(rowIndexes, firstKey, descending);
+        await validateByColumn(rowIndexes, secondKey, descending);
+      };
+      const sortingScenario = async (
+        firstSortBy: string,
+        firstSortOrder: string,
+        secondSortBy: string,
+        secondSortOrder: string,
+        scenario: () => Promise<void>,
+      ): Promise<void> => {
+        await test.step(`Sort by ${firstSortBy} and ${secondSortBy}`, async () => {
+          await multiSortMenu.open();
+          await multiSortMenu.multiSort.reset.pwLocator.click();
+          await multiSortMenu.close();
+          await multiSortMenu.open();
+
+          const firstRow = multiSortMenu.multiSort.rows.nth(0);
+          await firstRow.column.selectMenuOption(firstSortBy);
+          await firstRow.order.selectMenuOption(firstSortOrder);
+
+          await multiSortMenu.multiSort.add.pwLocator.click();
+
+          const secondRow = multiSortMenu.multiSort.rows.nth(1);
+          await secondRow.column.selectMenuOption(secondSortBy);
+          await secondRow.order.selectMenuOption(secondSortOrder);
+
+          await multiSortMenu.close();
+          await waitTableStable();
+          await scenario();
+        });
+      };
+
+      return { checkTableOrder, sortingScenario };
+    };
+
+    test.beforeAll(async ({ newProject }) => {
+      // create a new experiment for comparing Searcher Metric and Trial Count
+      await detExecSync(
+        `experiment create ${fullPath('examples/tutorials/core_api_pytorch_mnist/checkpoints.yaml')} --paused --project_id ${newProject.response.project.id}`,
+      );
+    });
+
+    // set table columns to have just the columns that we are using in the test cases.
+    test.beforeEach(async ({ authedPage, newProject }) => {
+      projectDetailsPage = new ProjectDetails(authedPage);
+      await projectDetailsPage.gotoProject(newProject.response.project.id);
+      const columnPicker = projectDetailsPage.f_experimentList.tableActionBar.columnPickerMenu;
+      await columnPicker.open();
+      await columnPicker.columnPickerTab.showAll.pwLocator.click();
+      await expect.soft(columnPicker.columnPickerTab.showAll.pwLocator).toHaveText('Hide all');
+      await columnPicker.columnPickerTab.showAll.pwLocator.click();
+
```

Comment:
this causes the test to stall as the soft expectation fails. actually check the value with an if statement to see if we have to click once or twice
```suggestion
      const showAllButton = columnPickerTab.showAll.pwLocator;
      await showAllButton.click();
      if (await showAllButton.textContent() === 'Hide all') {
        await showAllButton.click();
      }
```